profiling/src/profile/mod.rs

```diff
@@ -58,6 +57,38 @@ impl From<&SampleId> for u64 {
 #[repr(transparent)]
 pub struct StackTraceId(usize);
 
+#[derive(Copy, Clone, Default, Debug, Eq, PartialEq, Hash)]
+#[repr(transparent)]
+pub struct LocationId(u32);
+
+impl LocationId {
+    pub fn new<T>(v: T) -> Self
+    where
+        T: TryInto<u32>,
+        T::Error: Debug,
+    {
+        let index = v
+            .try_into()
+            .expect("the machine to run out of memory far before this happens");
+
+        // PProf reserves location 0.
+        // Both this, and the serialization of the table, add 1 to avoid the 0 element
+        Self(index + 1)
+    }
+}
+
+impl From<LocationId> for u64 {
+    fn from(s: LocationId) -> Self {
+        s.0.try_into().unwrap()
```

Comment:
Delegate to the implementation.

```suggestion
        Self::from(&s)
```

Reply:
Done